### PR TITLE
CONSOLE-3944: Disable segment analytics when cluster telemetry is disabled

### PR DESCRIPTION
**Pull request by TheRealJon**
Disable console-telemetry-plugin segment analytics when `window.SERVER_FLAGS.telemetry.TELEMETER_CLIENT_DISABLED` flag is set

### CODE REVIEW

**Comment by TheRealJon** [comment]
QE Approver
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @RickJWagner

File: frontend/packages/console-telemetry-plugin/src/listeners/segment.ts
Comment (yapei) [inline frontend/packages/console-telemetry-plugin/src/listeners/segment.ts:113]: Tested the changes and it looks like we have some issue here
1. confirm if telemeter client is disabled or not(based on latest code change, we check deployment/telemeter-client status)
```
$ oc get deployment telemeter-client -n openshift-monitoring
NAME               READY   UP-TO-DATE   AVAILABLE   AGE
telemeter-client   1/1     1            1           3h7m
$ oc get cm console-config -n openshift-console | grep tele (nothing returned)

```
2. Enable demo plugin and do some navigation in the web, we can see messages in browser console
```
 console-telemetry-plugin: telemetry disabled - ignoring telemetry event xxxx 
```

based on the data in step1 we can see that telemeter client is `NOT disabled`, so it looks a bit wired we are printing `telemetry disabled` message

Comment (yapei) [reply frontend/packages/console-telemetry-plugin/src/listeners/segment.ts:113]: after I disabled telemeter client, browser console prints following message(when telemeter is actually disabled, we didn't print correct message)
Screenshot 2024-04-07 at 2 31 31 PM

Comment (TheRealJon) [reply frontend/packages/console-telemetry-plugin/src/listeners/segment.ts:113]: There may be some confusion about the scope of this change. This PR is only meant to disable Segment analytics when cluster telemetry is disabled in the cluster monitoring operator. It will not disable all telemetry in the UI. 

Additionally, other conditions will cause Segment analytics to be disabled. Could you check these four conditions to ensure we shouldn't show the disabled message:
- window.SERVER_FLAGS?.telemetry?.DISABLED === 'false'
- window.SERVER_FLAGS?.telemetry?.DEVSANDBOX_DISABLED === 'false'
- window.SERVER_FLAGS?.telemetry?.TELEMETER_CLIENT_DISABLED === 'false'
- window.SERVER_FLAGS?.telemetry?.DEVSANDBOX_SEGMENT_API_KEY or window.SERVER_FLAGS?.telemetry?.SEGMENT_API_KEY is defined

**Review by jerolimov — APPROVED** [review]
Verified this change on a cluster bot:

Works as expected. I personally would recommend a small change on the operator side:

Allow cluster admins to override TELEMETRY_CLIENT_DISABLED with an annotation! Currently, there is no option to enable segment analytics if the `telemeter-client` Deployment isn't available because TELEMETRY_CLIENT_DISABLED is then always true.

@jhadvig maybe you can apply that if you touch the operator again.

Here are two long recordings how I've tested the change:

https://github.com/openshift/console/assets/139310/dad5e74e-ae9b-4528-b01b-93560e8c9af3

https://github.com/openshift/console/assets/139310/20e10ddf-444a-4a86-952b-68a5d6a6ccea

/lgtm

File: frontend/packages/console-telemetry-plugin/src/listeners/segment.ts
Comment (jerolimov) [reply frontend/packages/console-telemetry-plugin/src/listeners/segment.ts:113]: Hi @yapei , please take a look how I've tested the changes below. Maybe this is enough to get your qe-approve label.

The problem I've noticed with the `telemeter-client` Deployment is, that it was automatically removed again. I expect that the monitoring-operator reconciler removed it. A better test might be to enable telemetry on a cluster bot but I don't know how to enable it.

Comment (jerolimov) [reply frontend/packages/console-telemetry-plugin/src/listeners/segment.ts:113]: Oh, that was easier then expected. Instead of creating the `telemeter-client` Deployment manually, I just enabled telemetry in the `cluster-monitoring-config` ConfigMap in the `openshift-monitoring` namespace.

https://access.redhat.com/documentation/de-de/openshift_container_platform/4.1/html/telemetry/opting-out-of-telemetry

https://github.com/openshift/console/assets/139310/937e1d5e-6546-4942-87dd-a6be69525de4

Comment (TheRealJon) [inline frontend/packages/console-telemetry-plugin/src/listeners/segment.ts:113]: There might also be some confusion about this conditional block because there is no case where it will evaluate to true if TELEMETRY_DISABLED === true since this entire extension will not be resolved in that case.

Comment (TheRealJon) [inline frontend/packages/console-telemetry-plugin/src/listeners/segment.ts:113]: I don't want to add further changes to this PR since it's already got an lgtm, but I might follow on with a tweak like this:
```suggestion
  if (!apiKey) {
    // eslint-disable-next-line no-console
    console.debug(
      'console-telemetry-plugin: segment API key missing - ignoring telemetry event:',
      eventType,
      properties,
    );
    return;
```

**Comment by yapei** [comment]
Thanks @jerolimov @TheRealJon  except [this issue](https://github.com/openshift/console/pull/13677#discussion_r1567929201) everything looks good in my testing
/label qe-approved

**Pull request merged**
